bats/tests/helpers/vm.bash

```diff
@@ -39,6 +39,80 @@ flush_iptables() {
     wsl sudo iptables -X
 }
 
+# Helper to eject all existing ramdisk instances on macOS
+macos_eject_ramdisk() {
+    local mount="$1"
+    run hdiutil info -plist
+    assert_success
+    # shellcheck disable=2154 # $output set by `run`
+    run plutil -convert json -o - - <<<"$output"
+    assert_success
+    # shellcheck disable=2016 # $mount is interpreted by jq, not shell.
+    run jq_output --arg mount "$mount" \
+        '.images[]."system-entities"[] | select(."mount-point" == $mount) | ."dev-entry"'
+    assert_success
+    if [[ -z $output ]]; then
+        return
+    fi
+    # We don't need to worry about splitting here, it's all /dev/disk*
+    # However, we do need to ensure $output isn't clobbered.
+    # shellcheck disable=2206
+    local disks=($output)
+    local disk
+    for disk in "${disks[@]}"; do
+        CALLER="$(calling_function)" trace "$(umount "$disk" 2>&1 || :)"
+    done
+    for disk in "${disks[@]}"; do
+        CALLER="$(calling_function)" trace "$(hdiutil eject "$disk" 2>&1 || :)"
+    done
+}
+
+# Set up the use of a ramdisk for application data, to make things faster.
+setup_ramdisk() {
+    if ! using_ramdisk; then
+        return
+    fi
+
+    # Force eject any existing disks.
+    if is_macos; then
+        # Try to eject the disk, if it already exists.
+        macos_eject_ramdisk "$LIMA_HOME"
+    fi
+
+    local ramdisk_size="${RD_RAMDISK_SIZE}"
+    if ((ramdisk_size < ${RD_FILE_RAMDISK_SIZE:-0})); then
+        {
+            printf "RD:   %s requires %dGB of ramdisk; disabling ramdisk for this file\n" \
+                "$BATS_TEST_FILENAME" "$RD_FILE_RAMDISK_SIZE"
+        } >>"$BATS_WARNING_FILE"
+        printf "# WARN: %s requires %dGB of ramdisk; disabling ramdisk for this file\n" \
+            "$BATS_TEST_FILENAME" "$RD_FILE_RAMDISK_SIZE" >&3
```

Comment:
Looks like duplicated code. Maybe run it once and then print it twice?

```
run printf ...
assert_success
echo "$output" >>"$BATS_WARNING_FILE"
echo "$output" >&3
```

Not sure about calling `assert_success`.

Reply:
I opted to keep `assert_success` because it should pass.